http2_general/test_h2_stream_priority.py

```diff
@@ -0,0 +1,769 @@
+
+        client.send_bytes(
+            PriorityFrame(
+                stream_id=client.stream_id, depends_on=5, stream_weight=255, exclusive=False
+            ).serialize()
+        )
+        client.send_bytes(
+            PriorityFrame(
+                stream_id=client.stream_id + 2, depends_on=5, stream_weight=255, exclusive=False
+            ).serialize()
+        )
+        self.assertTrue(client.wait_for_connection_close())
+        self.assertIn(ErrorCodes.PROTOCOL_ERROR, client.error_codes)
+
+
+class TestBigHeadersAndBodyForSeveralStreams(TestPriorityBase, NetWorker):
+    def test_big_headers_and_body(self):
+        client, server = self.setup_test_priority(
+            extra_header=f"qwerty: {'y' * BIG_HEADER_SIZE}\r\n"
+        )
+        client.send_settings_frame(max_header_list_size=BIG_HEADER_SIZE * 2)
+        client.wait_for_ack_settings()
+
+        self.run_test_tso_gro_gso_def(
+            client,
+            server,
+            self._test_big_headers_and_body,
+            DEFAULT_MTU,
+        )
+
+    def _test_big_headers_and_body(self, client, server):
+        for i in range(1, 3):
+            client.make_request(self.post_request)
+
+        self.wait_for_responses(client)
+        self.check_response_sequence(client, 2)
+
+
+class TestSameWeightsWithOneParent(TestPriorityBase, NetWorker):
+    def test_content_type_not_progessive(self):
+        """
+        Check that we don't use raund robin for non progressive
+        streams and don't switch between such streams, when we
+        send data to client.
+        """
+        client, server = self.setup_test_priority(extra_header=f"content-type: image/png\r\n")
+
+        self.run_test_tso_gro_gso_def(
+            client,
+            server,
+            self._test_same_weights_streams_not_progressive,
+            DEFAULT_MTU,
+        )
+
+    def _test_same_weights_streams_not_progressive(self, client, server):
+        for i in range(1, 4):
+            client.make_request(
+                self.post_request,
+                end_stream=True,
+                priority_weight=16,
+                priority_depends_on=0,
+                priority_exclusive=False,
+            )
+            # Need to sure that tempesta receive requests in the correct order.
+            wair_for_headers()
+
+        """
+        Increment initial_window_size to max value, to prevent its affecting
+        on data sending (We switch to another stream if current stream exceeds
+        http window).
+        """
+        self.wait_for_responses(client, initial_window_size=2**31 - 1)
+        """
+        We send headers for all streams, and then blocked untlil initial
+        window size will be updated. Since the last stream has id == 5,
+        we start from it and don't switch to another stream until all data
+        will be sent or we exceeded HTTP window for this stream.
+        """
+        self.check_response_sequence(client, 3, [5, 1, 3])
```

Comment:
Do we have a test for trying to build a dependency cycle?

Reply:
There are several appropriate tests in h2spec